pkg/abaputils/abaputils.go

```diff
@@ -0,0 +1,254 @@
+package abaputils
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+)
+
+// GetAbapCommunicationArrangementInfo function fetches the communcation arrangement information in SAP CP ABAP Environment
+func GetAbapCommunicationArrangementInfo(options AbapEnvironmentOptions, c command.ExecRunner, oDataURL string, cfLogoutOption bool) (ConnectionDetailsHTTP, error) {
+
+	var connectionDetails ConnectionDetailsHTTP
+	var error error
+
+	if options.Host != "" {
+		// Host, User and Password are directly provided -> check for host prefix (double https)
+		match, err := regexp.MatchString(`^[hH][tT][tT][pP][sS]:\/\/.*`, options.Host)
```

Comment:
[minor] still not convinced if we need that funny check for a mix of upper case lower case in the schema. From the specs point of view lower case should be sufficient. Other option would be to check for `^(http|HTTP):\/\/.*`. But I know we have that already otherwise ...

Reply:
^(http|HTTP):\/\/.* is now used